mixer/cmd/reset.go

```diff
@@ -0,0 +1,200 @@
+			b.UpstreamVerUint32, err = parseUint32(b.UpstreamVer)
+			if err != nil {
+				fail(errors.Wrapf(err, "Couldn't parse upstream version"))
+			}
+			vFile := filepath.Join(b.Config.Builder.VersionPath, b.UpstreamVerFile)
+			if err = ioutil.WriteFile(vFile, []byte(b.UpstreamVer), 0644); err != nil {
+				fail(err)
+			}
+		}
+	}
+
+	// Change mixVersion to point to the last good build
+	if err := ioutil.WriteFile(filepath.Join(b.Config.Builder.VersionPath, b.MixVerFile), []byte(b.State.Mix.PreviousMixVer), 0644); err != nil {
+		fail(err)
+	}
+	b.MixVer = b.State.Mix.PreviousMixVer
+	b.MixVerUint32, err = parseUint32(b.MixVer)
+	if err != nil {
+		fail(errors.Wrapf(err, "Couldn't parse mixer version"))
+	}
+
+	// Make sure update/image/LAST_VER points to the last good build
+	if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "image", "LAST_VER"), []byte(b.State.Mix.PreviousMixVer), 0644); err != nil {
+		failf("Couldn't update LAST_VER file: %s", err)
+	}
+
+	// update the state.mix file
+	err = b.State.Save()
+	if err != nil {
+		failf("Couldn't update mixer.state file: %s", err)
+	}
+
+	// update www/version/latest_version file and check if sig file exists, if it does, regenerate the signature
+	// update www/version/formatXX/latest file and update the content, if signature exits , regenerate the signature
+	var skipSign bool
+	latestVerSigFilePath := filepath.Join(b.Config.Builder.ServerStateDir, "www", "version", "latest_version.sig")
+	if _, err = ioutil.ReadFile(latestVerSigFilePath); err != nil {
+		skipSign = true
+	}
+	err = b.UpdateLatest(skipSign)
+	if err != nil {
+		fail(err)
+	}
+
+	// if clean flag set
+	if resetCmdFlags.clean {
+		// Remove any folder inside update/image for versions above prevMixVer
+		files, err := ioutil.ReadDir(b.Config.Builder.ServerStateDir + "/image")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		for _, f := range files {
+			dirNameInt, err := parseUint32(f.Name())
+			if err != nil {
+				continue
+			}
+			if dirNameInt > stateMixPreviousMixVersionUint32 {
+				err := os.RemoveAll(b.Config.Builder.ServerStateDir + "/image/" + f.Name())
+				if err != nil {
+					continue
+				}
+			}
+		}
+
+		// Remove any folder inside update/www for versions above prevMixVer
+		files, err = ioutil.ReadDir(b.Config.Builder.ServerStateDir + "/www")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		for _, f := range files {
+			dirNameInt, err := parseUint32(f.Name())
+			if err != nil {
+				continue
+			}
+			if dirNameInt > stateMixPreviousMixVersionUint32 {
+				err := os.RemoveAll(b.Config.Builder.ServerStateDir + "/www/" + f.Name())
+				if err != nil {
+					continue
+				}
+			}
+		}
+
+		// Remove any folder inside update/www/version/ for version above prevStableVersion
+		files, err = ioutil.ReadDir(b.Config.Builder.ServerStateDir + "/www/version")
+		if err != nil {
+			log.Fatal(err)
+		}
+		// iterate over all the formats folder
+		for _, f := range files {
```

Comment:
this is also tricky for the same reason as stated for `latest_version`. I believe it is better to not change this folder.

Reply:
I am just reading files in this folder, not changing any files. These folders will never be used if clean flag is set. Is  this still tricky and break smtg?

Reply:
I'll comment on this issue on your revised code.